src/components/inputs/rich-text-input/plugins/undo.js

```diff
@@ -0,0 +1,47 @@
+import { isKeyHotkey } from 'is-hotkey';
+import memoize from 'lodash/memoize';
+
+const memoizedIsHotkey = memoize(isKeyHotkey);
+
+const HOT_KEY = 'mod+z';
+
+const hasUndos = editor => {
+  const { value } = editor;
+  const { data } = value;
+  const undos = data.get('undos');
+  return undos && undos.size > 1;
+};
+
+const UndoPlugin = (options = {}) => {
+  const hotkey = options.hotkey || HOT_KEY;
+  const isUndoHotkey = memoizedIsHotkey(hotkey);
+
+  return [
+    {
+      // eslint-disable-next-line consistent-return
+      onKeyDown(event, editor, next) {
+        if (!isUndoHotkey(event) || !hasUndos(editor)) {
+          return next();
+        }
+
+        event.preventDefault();
+        editor.undo();
+      },
+      queries: {
+        hasUndos: editor => hasUndos(editor),
+      },
+      commands: {
+        toggleUndo: editor => {
+          if (!editor.value.selection.isFocused) {
+            editor.focus();
+          }
+          if (hasUndos(editor)) {
+            editor.undo();
+          }
+        },
+      },
+    },
+  ];
+};
+
+export default UndoPlugin;
```

Comment:
How do you intend to test these plugins?

Reply:
I idea was just to test the whole component using puppeteer.